server/plugins/renderer/renderer.module.js

```diff
@@ -175,7 +179,10 @@ internals.parseResponse = async (bcAppData, request, response, responseArgs) =>
         request,
         stencilOptions: { get_data_only: true },
         stencilConfig: internals.getResourceConfig(bcAppData, request, configuration),
-        extraHeaders: { host: storeUrlObj.host },
+        extraHeaders: {
+            host: storeUrlObj.host,
+            stencil_custom_templates,
```

Comment:
I think you can move the stringify and fallback logic here, then you don't need to change anything in the  `buildReqHeaders`

Reply:
`buildReqHeaders` method is used in 2 places therefore I moved logic inside the method just to avoid duplication.